fuse-prometheus-operator.yml

```diff
@@ -0,0 +1,328 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: prometheus-operator
+  labels:
+    app: prometheus-operator
+  annotations:
+    openshift.io/display-name: "Red Hat Fuse prometheus-operator install"
+    openshift.io/provider-display-name: "Red Hat, Inc."
+    description: "."
+    tags: "fuse,prometheus,prometheus-operator,monitoring"
+    iconClass: "icon-rh-integration"
+    version: "1.0"
+message: |-
+  prometheus-operator is now deployed to ${NAMESPACE}
+parameters:
+- name: NAMESPACE
+  displayName: Namespace
+  value: fuse
+  required: true
+  description: Namespace in which the prometheus-operator is installed. 
+- name: CPU_REQUEST
+  displayName: CPU request
+  value: '0.2'
+  required: true
+  description: The amount of CPU to request for each container.
+- name: MEMORY_REQUEST
+  displayName: Memory request
+  value: 64Mi
+  required: true
+  description: The amount of memory required for each container to run.
+- name: CPU_LIMIT
+  displayName: CPU limit
+  value: '1.0'
+  required: true
+  description: The amount of CPU each container is limited to use.
+- name: MEMORY_LIMIT
+  displayName: Memory limit
+  value: '256Mi'
+  required: true
+  description: The amount of memory each container is limited to use.
+- name: CONTAINER_PORT
+  displayName: Memory limit
+  value: '8080'
+  required: true
+  description: The amount of memory each container is limited to use.
+- name: PROMETHEUS_MEMORY
+  displayName: Prometheus Memory Limit
+  value: '400Mi'
+- name: PROMETHEUS_SERVICE_PORT
+  displayName: Prometheus Service Port
+  value: '9090'
+- name: ENDPOINT_PORT
+  displayName: Endpoint Port
+  value: 'web'
+  required: true
+- name: PORT_NAME
+  displayName: Port Name
+  value: 'web'
+  required: true
+- name: FUSE_SERVICE_TEAM
+  displayName: Fuse Service team label
+  value: 'frontend'
+  requited: true
+
+
+objects:
+#
+# OpenShift resources 
+#
+- apiVersion: apiextensions.k8s.io/v1beta1
```

Comment:
@simonpasquier I was under the impression that the namespaced versions of the CRDs were necessary because of permissions, but I'll give it a try and see if it is really necessary  https://github.com/coreos/prometheus-operator/issues/1824

Reply:
IIUC custom resources **definitions** aren"t namespaced while the custom resources can be.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#customresourcedefinitionspec-v1beta1-apiextensions-k8s-io

Reply:
Is the [cluster-monitoring-operator](https://github.com/openshift/cluster-monitoring-operator) (CMO) deployed? When I tested the template, the CRDs were already created because CMO was deployed so my guess is that you don't run CMO. With OKD 3.11 CMO should be there by default so it really boils down to which OpenShift version you target?
If it is >= 3.11 then the template should assume that CRDs are present otherwise it should take care of creating them (maybe have a separate template for this?).

Reply:
Created separate config for the CRDs